### PR TITLE
Fix update logic after store properties on in-edges.

### PR DESCRIPTION
**Pull request by dangleptr**
Now after updating the out-edges successfully,  we update in-edges.

### CODE REVIEW

File: src/graph/UpdateEdgeExecutor.cpp [hunk 1]
```diff
@@ -279,11 +247,11 @@ void UpdateEdgeExecutor::execute() {
...
         }
-        if (insertable_ && rpcResp.get_upsert()) {
-            // TODO(zhangguoqing) Making the reverse edge of insertion is transactional
-            this->insertReverselyEdge(std::move(rpcResp));
+        if (reversely) {
+            doFinish(Executor::ProcessControl::kNext);
         } else {
-            this->finishExecution(std::move(rpcResp));
+            this->toResponse(std::move(rpcResp));
+            this->updateEdge(true);
```
Comment (bright-starry-sky) [inline src/graph/UpdateEdgeExecutor.cpp]: 👍

File: src/graph/UpdateEdgeExecutor.cpp [hunk 2]
```diff
@@ -190,68 +190,36 @@ void UpdateEdgeExecutor::finishExecution(storage::cpp2::UpdateResponse &&rpcResp
...
     resp = std::move(*resp_);
 }
 
-
-void UpdateEdgeExecutor::insertReverselyEdge(storage::cpp2::UpdateResponse &&rpcResp) {
-    std::vector<storage::cpp2::Edge> edges;
-    storage::cpp2::Edge reverselyEdge;
-    reverselyEdge.key.set_src(edge_.dst);
-    reverselyEdge.key.set_dst(edge_.src);
-    reverselyEdge.key.set_ranking(edge_.ranking);
```
Comment (laura-ding) [inline src/graph/UpdateEdgeExecutor.cpp]: why not set ranking

Comment (dangleptr) [reply src/graph/UpdateEdgeExecutor.cpp]: The edge_ is reused here.  The ranking has been set in the previous step.

File: src/graph/UpdateEdgeExecutor.cpp [hunk 3]
```diff
@@ -190,68 +190,36 @@ void UpdateEdgeExecutor::finishExecution(storage::cpp2::UpdateResponse &&rpcResp
...
+    std::string filterStr = "";
+    std::vector<std::string> returns;
+    if (reversely) {
+        auto src = edge_.src;
+        auto dst = edge_.dst;
+        edge_.set_src(dst);
+        edge_.set_dst(src);
+        edge_.set_edge_type(-edge_.edge_type);
+    } else {
+        filterStr = filter_ ? Expression::encode(filter_) : "";
```
Comment (darionyaphet) [inline src/graph/UpdateEdgeExecutor.cpp]: So the filter just processing when the edge is not reverse ?

Comment (dangleptr) [reply src/graph/UpdateEdgeExecutor.cpp]: When updating the reverse edges, it means the out edges have been updated succeeded. So we don't need to check the filter again.

File: src/storage/mutate/UpdateEdgeProcessor.cpp [hunk 4]
```diff
@@ -268,13 +270,13 @@ std::string UpdateEdgeProcessor::updateAndWriteBack(PartitionID partId,
...
                 if (!val_.empty()) {
                     if (rReader == nullptr) {
                         rReader = RowReader::getEdgePropReader(this->schemaMan_,
                                                                val_,
                                                                spaceId_,
-                                                               edgeKey.edge_type);
+                                                               std::abs(edgeKey.edge_type));
                     }
                     auto rValues = collectIndexValues(rReader.get(),
                                                       index->get_fields());
```
Comment (bright-starry-sky) [inline src/storage/mutate/UpdateEdgeProcessor.cpp]: Let me handle the in-edge prop of index with insert edge together.
Could you help change back line 273, 279 and 295?

Comment (dangleptr) [reply src/storage/mutate/UpdateEdgeProcessor.cpp]: Let me check the edge_type, if it is negative, just skip directly.  You could handle it later.

Comment (bright-starry-sky) [reply src/storage/mutate/UpdateEdgeProcessor.cpp]: > Let me check the edge_type, if it is negative, just skip directly. You could handle it later.

excellent!

**Comment by bright-starry-sky** [comment]
Well done!

**Pull request merged**